### PR TITLE
Add RPC API docs section, write eth_call reference

### PR DESCRIPTION
**Pull request by karalabe**
This PR adds an `JSON RPC APIs` section to the docs with a single document for `eth_call`. I've also added a bit of styling for tables, but we should make a followup PR for striped rows as well as to enable syntax highlighting. Those were working for the old Guide, the rework never implemented these.

### CODE REVIEW

File: docs/_rpc/eth_call.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+```
+
+##### 2. `Quantity | Tag` - Block number or the string `latest` or `pending`
+
+The *block number* is mandatory and defines the context (state) against which the specified transaction should be executed. It is not possible to execute calls against reorged blocks; or blocks older than 128 (unless the node is an archive node).
+
+##### 3. `Object` - Optional - State override set
```
Comment (FrankSzendzielarz) [inline docs/_rpc/eth_call.md]: Remove 'optional' as the opt/mandatory state is not used on the other fields

File: docs/_rpc/eth_call.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+| Field | Type | Bytes | Optional | Description |
+|:---:|:---:|:---:|:---:|:---|
+| `balance` | `Quantity` | <32 | Yes | Fake balance to set for the account before executing the call. |
+| `nonce` | `Quantity` | <8 | Yes | Fake nonce to set for the account before executing the call. |
+| `code` | `Binary` | any | Yes | Fake EVM bytecode to inject into the account before executing the call. |
+| `state` | `Object` | any | Yes | Fake key-value mapping to override **all** slots in the account storage before executing the call. |
+| `stateDiff` | `Object` | any | Yes | Fake key-value mapping to override **individual** slots in the account storage before executing the call. |
+
+The goal of the *state override set* is manyfold:
```
Comment (FrankSzendzielarz) [inline docs/_rpc/eth_call.md]: manifold not manyfold

File: docs/_rpc/eth_call.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    uint processConfirms;
+    uint threshold;
+
+    function VotingThreshold() public view returns (uint) {
+        return threshold;
+    }
+}
+```
+
+With a synced Rinkeby node with RPC exposed on localhost (`geth --rinkeby --rpc`) we can make a call against the live [Checkpoint Oracle](https://rinkeby.etherscan.io/address/0xebe8efa441b9302a0d7eaecc277c09d20d684540), but override its byte code with our own version that has an accessor for the voting theshold field:
```
Comment (ligi) [inline docs/_rpc/eth_call.md]: theshold -> threshold

File: docs/_rpc/eth_call.md (hunk 2 above)
Comment (karalabe) [reply docs/_rpc/eth_call.md]: Manifold (noun)
A copy made by the manifold writing process.

Manyfold (adverb)
By many times.

File: docs/_rpc/eth_call.md (hunk 3 above)
Comment (karalabe) [reply docs/_rpc/eth_call.md]: Thx, fixed upstream.

File: docs/_rpc/eth_call.md (hunk 1 above)
Comment (karalabe) [reply docs/_rpc/eth_call.md]: Thx, fixed upstream.

**Pull request merged**